### ISSUE TITLE
Improving kalman implementation flowdeck

### ISSUE THREAD

**Issue by knmcguire** [bitcraze/crazyflie-firmware#608]
Currently the flowdeck implementation for the kalman filter has a fixed standard deviation, no matter how much texture it sees. Moreover, it also sends flow measurements to the kalmanfilter while the sensor is actually giving a notice it is not seeing any motion at all (motion.motion). 

The idea is, is to do some measurements and look at the amount of features the flowdeck detects (squal) affects the STD at all to make an adaptive formula that gives a higher STD with lower texture. Also, the measurement should not be send to the kalmanfilter if the sensor indicates that there is no motion. That might also indicate why you see this wobbling effect due to the rotation compensation (no motion with rotation around x -y axis, will add 'compensation motion' to the measurment model with no actual motion to counteract it).

Anyway, hopefully these improvements will improve the flight with the flowdeck drastically.

**Comment by knmcguire** [comment]
Some nice plots to look at:
![image](https://user-images.githubusercontent.com/11061260/89533462-c330c700-d7f3-11ea-92a5-9e0fcf4be22a.png)
![image](https://user-images.githubusercontent.com/11061260/89533583-ee1b1b00-d7f3-11ea-831f-62f60e28399b.png)
On top is the detected motion of groundtruth in x and y, while flying a figure 8 on 0.5,1.0,1.5 and 2.0 meters with half low texture and half high texture surface. The bottom one is a scatter plot of the standard deviation (yaxis) vs the squal aka amount of features (xaxis), on which I based the formula on on the adative std in the flow deck now.

**Comment by knmcguire** [comment]
![image](https://user-images.githubusercontent.com/11061260/89907276-c14c7680-dbec-11ea-9cc7-1f9a95eec131.png)
So I noticed that there is a much clearer correlation between the shutter time and the standard deviation (see above plot: xaxis shutter time in uint16_t and yaxis is std). So I replaced the feature based on with this one.

To be completely honest, highering the STD already improved a lot, so the adaptive std is not adding that much extra value... but maybe it does in combination with other positioning systems?
